src/core/store/actions.ts

```diff
@@ -0,0 +1,195 @@
+  payload: skills,
+});
+export const resetEmployeesGrid = () => {
+  return {
+    type: actionTypes.RESET_EMPLOYEES_GRID,
+  };
+};
+
+//fetch methods
+export const fetchEmployeesData = (
+  params: {
+    limit: number;
+    offset: number;
+    sortBy: string;
+    sortDir: string;
+    search: string;
+    skillIds: string;
+  },
+  state: string
+) => {
+  return async function (dispatch: Dispatch) {
+    try {
+      dispatch(setLoading(actionTypes.SET_LOADING, true));
+      const response = await getData(apiURL.employee, { params: params });
+      const employeesResponseData = response.data.data;
+      const employees = employeesResponseData.employees;
+      if (state === "List")
+        dispatch(
+          setEmployeesForList({
+            ...employeesResponseData,
+            employees: employees.map(
+              (employee: {
+                id: string;
+                firstName: string;
+                lastName?: string;
+                isActive?: boolean;
+                dob?: string;
+                email?: string;
+                phone?: string;
+                designation?: string;
+                salary?: string;
+                dateOfJoining?: string;
+                address?: string;
+                moreDetails: string;
+                role?: IRole;
+                department?: IDepartment;
+                skills?: ISkill[];
+              }) => {
+                return {
+                  ...employee,
+                  moreDetails: employee.moreDetails
+                    ? JSON.parse(employee.moreDetails)
+                    : null,
+                };
+              }
+            ),
+          })
+        );
+      else
+        dispatch(
+          setEmployeesForGrid({
+            ...employeesResponseData,
+            employees: employees.map(
+              (employee: {
+                id: string;
+                firstName: string;
+                lastName?: string;
+                isActive?: boolean;
+                dob?: string;
+                email?: string;
+                phone?: string;
+                designation?: string;
+                salary?: string;
+                dateOfJoining?: string;
+                address?: string;
+                moreDetails: string;
+                role?: IRole;
+                department?: IDepartment;
+                skills?: ISkill[];
+              }) => {
+                return {
+                  ...employee,
+                  moreDetails: employee.moreDetails
+                    ? JSON.parse(employee.moreDetails)
+                    : null,
+                };
+              }
+            ),
+          })
+        );
+    } catch (error) {
+      toast.error("No data is recieved", { toastId: "no-data" });
+      console.error("Error fetching data:", error);
+    } finally {
+      dispatch(setLoading(actionTypes.SET_LOADING, false));
+    }
+  };
+};
+
+export const fetchDropdownData = () => {
```

Comment:
I guess here the fetch happens sequentially one after the other. We will have to make these fetches simultaneously.

Reply:
changed